api/internal/git/repospec.go

```diff
@@ -86,7 +87,10 @@ func NewRepoSpecFromURL(n string) (*RepoSpec, error) {
 	if filepath.IsAbs(n) {
 		return nil, fmt.Errorf("uri looks like abs path: %s", n)
 	}
-	host, orgRepo, path, gitRef, gitSubmodules, suffix, gitTimeout := parseGitURL(n)
+	host, orgRepo, path, gitRef, gitSubmodules, suffix, gitTimeout, err := parseGitURL(n)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This condition can be hit. Line 121-123 returns the non-nil error. `parseGitURL()` uses named returns, which imho is not best practice for this type of function. I didn't want to change the pattern in this PR, but I do believe the behavior is correct. See https://go.dev/ref/spec#Return_statements

Reply:
You are right. I think two improvements can be made to improve the code readability.
1. Do not use named returns (we avoid named returns in other sibling projects so I was in that assumption in the first review)
2. The return has too many variables (normally the number of return values are in 0 to 3 range, extending that is a signal that the code should be refactored in some way)

These are just readability tips but I agree with you that this PR can follow the pattern and focus on its own task. 
cc @annasong20 whom may have some better ideas.

Reply:
Nothing much to add! When @natasha41575 suggested the same thing [here](https://github.com/kubernetes-sigs/kustomize/pull/4652#discussion_r936926790), I returned a `RepoSpec` instead; it might be even easier for you since you're already returning an error. I don't feel strongly either way.

Reply:
I changed `parseGitURL` to return a `&RepoSpec`, though I did it in a rather quirky way. Instantiating a new one at every return site would add ~60 lines, and that doesn't seem like it would aid readability.